### PR TITLE
suppress expansion of unused raft spectral templates

### LINKED ISSUE #2678: spectral_clustering build is suspiciously slow and contributes to CI build timeouts
We need to look at why `spectral_clustering.cu` takes ~10x the time to compile vs. all others.  On my personal workstation I observed >6 minutes to build this file vs. <0.5 minutes for others.
We have been seeing several CI build timeouts lately for specific platforms, and the excessively long time to build spectral_clustering is believed to be a contributing factor.  Plus, local development from a clean build is impacted for every developer.

### PR DESCRIPTION
**Pull request by ChuckHastings**
Closes #2678 

raft updated some k-means logic and dramatically increased the compile time of our spectral clustering implementation.  This PR adds an include that will suppress the expansion of templates that we are not using.

### CODE REVIEW

**Comment by ChuckHastings** [comment]
rerun tests

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -146,6 +146,7 @@ include(${rapids-cmake-dir}/cpm/cuco.cmake)
 rapids_cpm_cuco(BUILD_EXPORT_SET cugraph-exports)
 
 include(cmake/thirdparty/get_raft.cmake)
+find_package(raft COMPONENTS distance)
```
Comment (cjnolet) [inline cpp/CMakeLists.txt:149]: `get_raft.cmake` should be doing this now so there shouldn't be a need for this line

Comment (ChuckHastings) [reply cpp/CMakeLists.txt]: Fixed

File: conda/recipes/libcugraph/meta.yaml [hunk 2]
```diff
@@ -63,6 +63,11 @@ outputs:
         - cmake {{ cmake_version }}
       run:
         - cudatoolkit {{ cuda_spec }}
+<<<<<<< HEAD
+        - libraft-headers {{ minor_version }}
+        - libraft-distance {{ minor_version }}
+=======
+>>>>>>> branch-22.10
```
Comment (rlratzel) [inline conda/recipes/libcugraph/meta.yaml:70]: Looks like a merge conflict was committed. This is why CI is failing.

**Review by rlratzel — CHANGES_REQUESTED** [review]

**Comment by ChuckHastings** [comment]
rerun tests

File: conda/recipes/libcugraph/meta.yaml [hunk 3]
```diff
@@ -38,9 +38,9 @@ requirements:
   host:
     - doxygen {{ doxygen_version }}
     - cudatoolkit {{ cuda_version }}.*
+    - libraft-headers {{ minor_version }}.*
+    - libraft-distance {{ minor_version }}.*
```
Comment (cjnolet) [inline conda/recipes/libcugraph/meta.yaml:42]: I think this might also need to be added to pylibcugraph and cugraph

Comment (ChuckHastings) [reply conda/recipes/libcugraph/meta.yaml]: Done in latest push

**Comment by cjnolet** [comment]
@vyasr do you have any ideas as to why this would still be failing when libraft-distance is installed as a dependency of cugraph/pylibcugraph? I'm scratching my head a little bit.

**Comment by ChuckHastings** [comment]
> @cjnolet @ChuckHastings If you're pulling libraft-distances then I believe the target that you should be linking to is `raft::raft_distance_lib` instance of `raft::distance`.

Trying that now, thanks!

**Comment by vyasr** [comment]
Hmm apparently not the issue? Can you reproduce this locally? I can give that a shot if you haven't been able to.

**Comment by ChuckHastings** [comment]
Corey fixed this in #2739 ,  we'll merge that PR instead.

**Comment by vyasr** [comment]
Ahhh I think I see the issue.

**Pull request closed without merging**